component-api/src/main/java/org/talend/sdk/component/api/record/Schema.java

```diff
@@ -71,10 +72,44 @@
     Stream<Entry> getAllEntries();
 
     /**
-     * @return schema builder from this schema.
+     * Get a Builder from the current schema.
+     *
+     * @return a {@link Schema.Builder}
      */
     Schema.Builder toBuilder();
 
+    /**
+     * Get all entries sorted by schema designed order.
+     *
+     * @return all entries ordered
+     */
+    List<Entry> getEntriesOrdered();
+
+    /**
+     * Get all entries sorted using a custom comparator.
+     *
+     * @param comparator the comparator
+     * @return all entries ordered with provided comparator
+     */
+
+    List<Entry> getEntriesOrdered(Comparator<Entry> comparator);
+
+    /**
+     * Get all entries sorted using a custom EntriesOrder .
+     *
+     * @param entriesOrder
+     * @return all entries ordered with provided EntriesOrder
+     */
+    List<Entry> getEntriesOrdered(EntriesOrder entriesOrder);
```

Comment:
As EntriesOrder class already implement Comparator<Entry>, this method is unuseful here.